src/mlpack/core/data/dataset_mapper.hpp

```diff
@@ -147,6 +147,15 @@ class DatasetMapper
    */
   size_t Dimensionality() const;
 
+  /**
+   * Remove a dimension and its corresponding mappings. Note that all dimensions
+   * after the deleted dimension will have their indices shifted forward by 1,
```

Comment:
```suggestion
   * after the deleted dimension will have their indices reduced by 1,
```
I know this is a little pedantic but I think this is just a bit clearer.